util/metrics/src/service_metrics.rs

```diff
@@ -95,14 +96,16 @@ impl ServiceMetrics {
         }
     }
 
-    /// Register service
+    /// Register Prometheus metrics family
     pub fn new_and_registered() -> ServiceMetrics {
         let svc = ServiceMetrics::default();
         let _res = prometheus::register(Box::new(svc.clone()));
         svc
     }
 
-    /// Track number of requests and durations
+    /// Takes the RpcContext used during a gRPC method call to get the method
+    /// name and increments counters tracking the number of calls to and
+    /// durations of that method
```

Comment:
nit: This feels like it's describing the inner workings of the function and not necessarily what a user wants/needs to know.

Perhaps:
```suggestion
    /// Increments the number of requests for the active method in `ctx`,
    /// returning a [Option<HistogramTimer>] to track the duration.
    /// The returned [Option<HistogramTimer>] is meant to be used as an RAII
    /// guard in order to accurately track the method duration.
```
Not sure on the format to correctly document the `HistogramTimer` perhaps the `Option` should be omitted from it in the docs.

I'm hoping someone knows a better term. RAII always felt like an unintuitive term, but I ran across this, https://rust-unofficial.github.io/patterns/patterns/behavioural/RAII.html

Reply:
I think just "returning a histogram timer to track the duration" I don't want get overly verbose.